### PR TITLE
Add MultiSelect component

### PR DESCRIPTION
**Pull request by javierarce**
screen shot 2016-07-05 at 09 57 00

```
      columns: {
        type: 'MultiSelect',
        title: 'what',
        options: this._columnOptions.filterByType(['string', 'number']),
        editorAttrs: {
          showSearch: true
        }
      }
```

CR: @xavijam

### CODE REVIEW

**Comment by saleiva** [comment]
OMG! :)

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-collection.js [hunk 1]
```diff
@@ -10,7 +10,6 @@ var _ = require('underscore');
  */
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-collection.js]: I see so much logic, why not create an extended custom-list-collection file?
multiple-choice-list-collection?

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ var _ = require('underscore');
 module.exports = CoreView.extend({
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js]: Same for this one, I'll create an extended item.

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-with-checkbox.tpl [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<button type="button" class="CDB-ListDecoration-itemLink u-ellipsis
+  <% if (isSelected) { %> is-selected <% } %>
+  <% if (isDestructive) { %>
+  u-errorTextColor
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-with-checkbox.tpl]: Don't we miss a <% } %>. It is a bit difficult to read.

**Comment by xavijam** [comment]
It looks great man! But in my opinion, I'd separate the custom-list implementation from the stuff you have built, because the custom-list thing was built like something super simple, easy customizable with the items. With this new logic, my feeling is we should create an extended component.

**Comment by javierarce** [comment]
I thought our current policy was duplicate over extend, but it's a simple change.

**Comment by xavijam** [comment]
Yeah, totally, but IMHO this element is getting more complexity than he should have.

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-with-checkbox.tpl (hunk 3 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-with-checkbox.tpl]: It should be fine. It's exactly what we have in custom-list-item.tpl plus the checkbox input.

**Comment by xavijam** [comment]
❤️ 💋 😍

**Comment by javierarce** [comment]
Ok, refactored. Thanks!

**Comment by xavijam** [comment]
ready to merge! :)

**Comment by javierarce** [comment]
Thanks!

**Pull request merged**